### ISSUE TITLE
Configurator Addon 3.0 on 0.91.2 - no config and port

### ISSUE THREAD

**Issue by woody4165** [danielperna84/hass-configurator#153]
I had to update to 0.91.2 from 0.91.0 to have Configurator 3.0 to install.

But now the Config field and Network host filed are empty and even if I copy from the doc page and save it does not remain.

![Schermata 2019-04-09 alle 09 08 15](https://user-images.githubusercontent.com/6501444/55780405-1c75d280-5aa8-11e9-8b07-1588448f9219.png)

I’m on Hassio installed in docker on Ubuntu using this to install

curl -sL "https://raw.githubusercontent.com/home-assistant/hassio-build/master/install/hassio_install" | bash -s

I’ve tried to uninstall and reinstall the addon, but no change

Any advice?

Thanks

**Comment by aluisperezh** [comment]
I can confirm this, running on Raspbian. I even tried to access the Docker container and modify settings inside, but it won't work.

**Comment by Marlor** [comment]
From what i understand, thanks to "Ingress", the config is not needed anymore.
The only thing missing is how to add it to an iframe.

**Comment by tofuSCHNITZEL** [comment]
probably has something to do with the addon using the HA ingress routing feature...
my configurator is available via https://*myhadomain*/hassio/ingress/core_configurator

**Comment by aluisperezh** [comment]
You are right, it's there. And when I try to "Open Web UI" from Hass.io menu it opens inside iframe.

Thanks and regards

**Comment by woody4165** [comment]
Now I've reverted back to 2.1 from a snapshot.
When on version 3.0 from the addon page clicking on OPEN WEB UI the Configurator was opening, but from http://mydomain:3218 was not reachable and I cannot define username/password since the Config does not remain

**Comment by tofuSCHNITZEL** [comment]
> 
> 
> Now I've reverted back to 2.1 from a snapshot.
> When on version 3.0 from the addon page clicking on OPEN WEB UI the Configurator was opening, but from http://mydomain:3218 was not reachable and I cannot define username/password since the Config does not remain

yes because it is not necessary for version 3.0 anymore because the configurator gui is served via HA so no extra configuration necessary anymore.

The part we need to figure out is how to Link items in the HA menu without opening them in an iframe

**Comment by woody4165** [comment]
> The part we need to figure out is how to Link items in the HA menu without opening them in an iframe

The answer is here
[https://community.home-assistant.io/t/configurator-addon-3-0-on-0-91-2-no-config-and-port/110312/7?u=woody4165](url)

```
panel_iframe:
configurator:
title: Configurator
icon: mdi:wrench
url: http://hassio_ip:8123/hassio/ingress/core_configurator
```

**Comment by aluisperezh** [comment]
I'm afraid that, at least in my case, that config opens another HA interface inside the iFrame...

**Comment by tofuSCHNITZEL** [comment]
> 
> 
> I'm afraid that, at least in my case, that config opens another HA interface inside the iFrame...

exactly. this displays another "full" HA instance in the frame.

**Comment by Marlor** [comment]
I got a solution:
OPEN WEB UI -> in the top right there is a "pop out" button.
Use the link from this button as your panel_iframe URL.
Should look like:
https://yourdomain/api/hassio_ingress/"longpw"

**Comment by tofuSCHNITZEL** [comment]
perfect! hopefully the last part is not dynamic?

**Comment by aluisperezh** [comment]
It seems not. I restarted HA and then Configurator and it still works.

Thanks!!

**Comment by Marlor** [comment]
After some reboots it stayed the same.
This is probertly not the right way to do it, but a workaround until the docs are up to date.

**Comment by woody4165** [comment]
The explanation to solve all the issues is in this [thread message](https://community.home-assistant.io/t/simplistic-configuration-ui/10175/334) where @danielperna84 explains all the news.

So I think it's time to close this issue

**Comment by danielperna84** [comment]
> So I think it's time to close this issue

Leave it open for a couple of days. Others may run into the same problems. If they see this issue they have a good source for solutions to their problems. 👍

**Comment by nhorvath** [comment]
This is an unfortunate change. I used to be able to only allow configurator access to specific IPs via firewall rules. Now anywhere I can access HA has access to configurator. Something I don't want.

**Comment by nitehawk98** [comment]
When I click the Open WebUI I get an 401: unauthorized after upgrading to 3.0

**Comment by aluisperezh** [comment]
I'm afraid that it has something to do with the "Store user?" (or something like that, my HA is in Spanish) the first time we accessed Configurator from HA interface. It happens the same to me in one computer, but not in other. I'm fighting with it right now :)

**Comment by nitehawk98** [comment]
Just tested it on a other computer and there it works without problems, weird

**Comment by nhorvath** [comment]
Try clearing your cookies and cache in the browser that is not working, it should make you log in again to home assistant and you might have better luck.

**Comment by nitehawk98** [comment]
Nope didn't work, still the same 401 error

**Comment by aluisperezh** [comment]
Try to open from hass.io->Configurator->Open Web UI first. Once there (it WILL open), it will open from HA menu. Why? I really don't know...

**Comment by nitehawk98** [comment]
already tried that one, on one computer I get the 401: Unauthorized error, on another computer it works fine.

**Comment by aluisperezh** [comment]
I tried in the same (non working before) computer with anothe browser, MS Edge, and it worked. I think it has something to do with the browser, but going through the add-on "Open Wen UI" first solved it in my Chrome. Sorry if it's not working for you :(

**Comment by nitehawk98** [comment]
Also tried another browser and then it works.
Must be a browser plugin or something that is blocking something. 

will investigate some more.

Thanks for the help.

**Comment by danielperna84** [comment]
> I used to be able to only allow configurator access to specific IPs via firewall rules. Now anywhere I can access HA has access to configurator. Something I don't want.

You can install the add-on from my repository at https://github.com/danielperna84/hassio-addons
The version you'll get there has all the options like you know them from previous verision. Obviously it lacks the ingress-feature, but you don't seem to need that anyways. 😉

**Comment by nhorvath** [comment]
> > I used to be able to only allow configurator access to specific IPs via firewall rules. Now anywhere I can access HA has access to configurator. Something I don't want.
> 
> You can install the add-on from my repository at https://github.com/danielperna84/hassio-addons
> The version you'll get there has all the options like you know them from previous verision. Obviously it lacks the ingress-feature, but you don't seem to need that anyways. 😉

Oh ok, that's a good option, but what's the difference between the one I have and your branch? Did it fork at some point? Just trying to understand where the 2 versions come from.

**Comment by danielperna84** [comment]
> Just trying to understand where the 2 versions come from.

At first there was only my repository. The hassio-repository was brand new back then with only a hand full of add-ons. It took a couple weeks for my add-on to be added to the official repository (for technical reasons, local building wasn't supported back then).
Once the add-on got included in the official repo, I stopped maintaining my repo because pretty much nobody was using it.
After a while though I have encountered a new _limitation_: whenever I upgraded the configurator, the add-on users had to wait until my pull request to update the add-on has been merged. In some cases that took several days, even weeks. The PR for the latest version even has been created in November last year. During that time 2 further versions have been released. So essentially hassio user were bound to the outdated version.
My solution: reviving my repo, which more or less did the same as the official repo, but I had control of when I wanted to release something new. Also I didn't strip out options I actually found useful, whereas they were stripped out in the official repo to make it easier to configure.

TL;DR
The version in my repo gets updated immediately when a new version gets released, but is aimed more at _advanced_ users in terms of the configuration possibilities.

**Comment by aluisperezh** [comment]
I've gone back to installing from Daniel's repository and everything seems to work fine. The method I posted (start first from hass.io menu) only works once. Whenever HA is restarted you have to go through the same process again...

I forgot to say THANKS for your work, Daniel. I could not live without it.

**Comment by woody4165** [comment]
So, if now I add [https://github.com/danielperna84/hassio-addons](https://github.com/danielperna84/hassio-addons) to the Addon Store, remove the Configurator from the std store and install your, I will get update as soon as you will publish, correct?

**Comment by ronmichel** [comment]
> I'm afraid that, at least in my case, that config opens another HA interface inside the iFrame...

Is there any way on preventing opening the iFrame in the second iFrame? I don't want to use the .../api/hassio_ingress/"token" in my config because that token is dynamic (at least dependent on HA installation)
I would prefere using /hassio/ingress/core_configurator but here the frame inside frame issue occurs.

**Comment by danielperna84** [comment]
> So, if now I add https://github.com/danielperna84/hassio-addons to the Addon Store, remove the Configurator from the std store and install your, I will get update as soon as you will publish, correct?

Correct. But I will not implement the ingress-feature that the official add-on has. So you have to do the manual configuration like it was before the new V3 add-on.

**Comment by tofuSCHNITZEL** [comment]
I'm now sucessfully using this config to have it linked in the sidbar:
https://github.com/home-assistant/hassio-addons/issues/562#issuecomment-482604487

**Comment by cogneato** [comment]
@nhorvath 

> since HA doesn't have any permission system for it's users yet

Any user assigned the role of "user" does not have access to configuration or the hassio panel at all. You have to be administrator. It is using SSO and when you open this new ingress version of configurator you are signing in with whatever admin you are unsing currently signed in to home assistant.

**Comment by nhorvath** [comment]
Thanks, yes I just noticed they introduced user levels.

**Comment by Marlor** [comment]
Good news everyone:
https://www.home-assistant.io/blog/2019/04/15/hassio-ingress/
In 0.92 we can enable a sidebar/panel_iframe button via toggle on the add-on details page.

**Comment by danielperna84** [comment]
Closing this issue now since most of the people should be up and running, and the switch to integrate the configurator into the UI is available too.
